src/reducers/copy.js

```diff
@@ -39,50 +39,81 @@ export default (pattern) => (base, source) => {
-        if (sourceField.tag === baseField.tag &&
-          sourceField.ind1 === baseField.ind1 &&
-          sourceField.ind2 === baseField.ind2 &&
-          sourceField.subfields.length === baseField.subfields.length) {
-          return baseField.subfields.every(isIdenticalSubfield);
-        }
-        function normalizeSubfield(subfield) {
-          return subfield.value.toLowerCase().replace(/\s+/u, '');
-        }
-        function isIdenticalSubfield(baseSub) {
-          const normBaseSub = normalizeSubfield(baseSub);
-          return sourceField.subfields.some(sourceSub => {
-            const normSourceSub = normalizeSubfield(sourceSub);
-            return normSourceSub === normBaseSub;
-          });
+          function normalizeControlField(field) {
+            return field.value.toLowerCase().replace(/\s+/u, '');
+          }
+
+          function isIdenticalControlField(baseField) {
+            const normalizedBaseField = normalizeControlField(baseField);
+            const normalizedSourceField = normalizeControlField(sourceField);
+            return normalizedSourceField === normalizedBaseField;
+          }
+          function isIdenticalDataField(baseField) {
+            if (sourceField.tag === baseField.tag &&
+              sourceField.ind1 === baseField.ind1 &&
+              sourceField.ind2 === baseField.ind2 &&
+              sourceField.subfields.length === baseField.subfields.length) {
+              return baseField.subfields.every(isIdenticalSubfield);
+            }
+            function normalizeSubfield(subfield) {
+              return subfield.value.toLowerCase().replace(/\s+/u, '');
+            }
+
+            function isIdenticalSubfield(baseSub) {
+              const normBaseSub = normalizeSubfield(baseSub);
+                // If excluded subfields have been defined for this field
+                if(excludeSubfields) {
+                  debug(`in excludeSubfields block`);
+                  const excludedSubfields = [...excludeSubfields]; // make array from string
+                  debug(`excludedSubfields: ${JSON.stringify(excludedSubfields, undefined, 2)} ${Array.isArray(excludedSubfields)}`);
+                  // Return the subfields that are not in excludedSubfields
+                  function filterExcluded(subfield) {
+                    return excludedSubfields.indexOf(subfield.code) === -1;
+                  }
+                  const includedSourceSubs = sourceField.subfields.filter(filterExcluded);
+                  debug(`includedSourceSubs: ${JSON.stringify(includedSourceSubs, undefined, 2)}`);
+                  return includedSourceSubs.some(sourceSub => {
+                    const normSourceSub = normalizeSubfield(sourceSub);
+                    return normSourceSub === normBaseSub;
+                  })
+                }
+                // If there are no excluded subfields, all subfields are compared
+                if(!excludeSubfields) {
```

Comment:
Not needed. You only need `if (excludeSubfields) {` and then return from that block